game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/SelectMainBattleCasualtiesTest.java

```diff
@@ -0,0 +1,526 @@
+            firingGroup,
+            fireRoundState,
+            (arg1, arg2) -> new CasualtyDetails());
+
+    final SelectMainBattleCasualties.Select selectFunction =
+        mock(SelectMainBattleCasualties.Select.class);
+    final CasualtyDetails expected = new CasualtyDetails(targetUnits, List.of(), false);
+    when(selectFunction.apply(any(), any(), anyCollection(), anyInt())).thenReturn(expected);
+
+    final CasualtyDetails details =
+        new SelectMainBattleCasualties(selectFunction).apply(delegateBridge, selectCasualties);
+
+    assertThat(details.getKilled().toArray(), is(targetUnits.toArray()));
+    // edit mode always sets auto calculated to true
+    assertThat(details.getAutoCalculated(), is(true));
+
+    verify(selectFunction)
+        .apply(
+            eq(delegateBridge),
+            eq(selectCasualties),
+            (Collection<Unit>) argThat(containsInAnyOrder(targetUnits.toArray())),
+            eq(0));
+  }
+
+  @Nested
+  class TransportCasualtiesNotRestricted {
+
+    @Test
+    void moreHitPointsThanHits() {
+      final List<Unit> targetUnits = List.of(givenAnyUnit(), givenAnyUnit());
+
+      targetUnits.forEach(
+          unit -> {
+            final UnitAttachment unitAttachment =
+                (UnitAttachment) unit.getType().getAttachment(UNIT_ATTACHMENT_NAME);
+            when(unitAttachment.getHitPoints()).thenReturn(2);
+          });
+
+      final FiringGroup firingGroup = new FiringGroup("", "", List.of(), targetUnits);
+
+      final FireRoundState fireRoundState = new FireRoundState();
+      fireRoundState.setDice(givenDiceRollWithHits(3));
+
+      final SelectCasualties selectCasualties =
+          new SelectCasualties(
+              battleState,
+              BattleState.Side.OFFENSE,
+              firingGroup,
+              fireRoundState,
+              (arg1, arg2) -> new CasualtyDetails());
+
+      final SelectMainBattleCasualties.Select selectFunction =
+          mock(SelectMainBattleCasualties.Select.class);
+      final CasualtyDetails expected = new CasualtyDetails();
+      when(selectFunction.apply(any(), any(), anyCollection(), anyInt())).thenReturn(expected);
+
+      final CasualtyDetails details =
+          new SelectMainBattleCasualties(selectFunction).apply(delegateBridge, selectCasualties);
+
+      assertThat(details, is(expected));
+
+      verify(selectFunction)
+          .apply(
+              eq(delegateBridge),
+              eq(selectCasualties),
+              (Collection<Unit>) argThat(containsInAnyOrder(targetUnits.toArray())),
+              eq(3));
+    }
+
+    @Test
+    void moreHitsThanHitPoints() {
+      final List<Unit> targetUnits = List.of(givenAnyUnit(), givenAnyUnit());
+
+      targetUnits.forEach(
+          unit -> {
+            final UnitAttachment unitAttachment =
+                (UnitAttachment) unit.getType().getAttachment(UNIT_ATTACHMENT_NAME);
+            when(unitAttachment.getHitPoints()).thenReturn(2);
```

Comment:
It looks like there are more hit points here than hits, is the name of the test correct? "moreHitsThanHitPoints"

Reply:
The unit has a total of 2 hit points but has already taken one hit.  So the unit has 1 hit point left.  The amount of hits in the test is 3 and there are two units.  So there are more hits than hit points.

I think this is another terminology problem.  Is "hit points" the amount the unit has ignoring any hits they've taken or is it the amount of the user has including the hits they've taken?

Reply:
Pretty sure - hit points is total hit points regardless of damage. Hit points left or HP remaining is hit points minus damage.